src/algebra/order/hom/monoid.lean

```diff
@@ -0,0 +1,481 @@
+@[simp, to_additive] lemma comp_apply (f : β →*o γ) (g : α →*o β) (a : α) :
+  (f.comp g) a = f (g a) := rfl
+@[simp, to_additive] lemma comp_to_monoid_hom (f : β →*o γ) (g : α →*o β) :
+  (f.comp g).to_monoid_hom = f.to_monoid_hom.comp g.to_monoid_hom := rfl
+@[simp, to_additive] lemma comp_to_order_hom (f : β →*o γ) (g : α →*o β) :
+  (f.comp g).to_order_hom = f.to_order_hom.comp g.to_order_hom := rfl
+@[simp, to_additive] lemma comp_assoc (f : γ →*o δ) (g : β →*o γ) (h : α →*o β) :
+  (f.comp g).comp h = f.comp (g.comp h) := rfl
+@[simp, to_additive] lemma comp_id (f : α →*o β) : f.comp (order_monoid_hom.id α) = f :=
+order_monoid_hom.ext $ λ a, rfl
+@[simp, to_additive] lemma id_comp (f : α →*o β) : (order_monoid_hom.id β).comp f = f :=
+order_monoid_hom.ext $ λ a, rfl
+
+@[to_additive]
+lemma cancel_right {g₁ g₂ : β →*o γ} {f : α →*o β} (hf : function.surjective f) :
+  g₁.comp f = g₂.comp f ↔ g₁ = g₂ :=
+⟨λ h, order_monoid_hom.ext $ hf.forall.2 $ order_monoid_hom.ext_iff.1 h, congr_arg _⟩
+
+@[to_additive]
+lemma cancel_left {g : β →*o γ} {f₁ f₂ : α →*o β} (hg : function.injective g) :
+  g.comp f₁ = g.comp f₂ ↔ f₁ = f₂ :=
+⟨λ h, order_monoid_hom.ext $ λ a, hg $
+  by rw [←order_monoid_hom.comp_apply, h, order_monoid_hom.comp_apply], congr_arg _⟩
+
+/-- `1` is the homomorphism sending all elements to `1`. -/
+@[to_additive "`1` is the homomorphism sending all elements to `1`."]
+instance : has_one (α →*o β) := ⟨{ monotone' := monotone_const, ..(1 : α →* β) }⟩
+
+
+@[simp, to_additive] lemma coe_one : ⇑(1 : α →*o β) = 1 := rfl
+@[simp, to_additive] lemma one_apply (a : α) : (1 : α →*o β) a = 1 := rfl
+
+@[simp, to_additive] lemma one_comp (f : α →*o β) : (1 : β →*o γ).comp f = 1 := rfl
+@[simp, to_additive] lemma comp_one (f : β →*o γ) : f.comp (1 : α →*o β) = 1 :=
+by { ext, exact map_one f }
+
+section mul
+
+/-- For two ordered monoid morphisms `f` and `g`, their product is the ordered monoid morphism
+sending `a` to `f a * g a`. -/
+@[to_additive "For two ordered additive monoid morphisms `f` and `g`, their product is the ordered
+additive monoid morphism sending `a` to `f a + g a`."]
+instance : has_mul (α →*o β) :=
+⟨λ f g, { monotone' := f.monotone'.mul' g.monotone', ..(f * g : α →* β) }⟩
+
+@[simp, to_additive] lemma coe_mul (f g : α →*o β) : ⇑(f * g) = f * g := rfl
+@[simp, to_additive] lemma mul_apply (f g : α →*o β) (a : α) : (f * g) a = f a * g a := rfl
+
+@[to_additive] lemma mul_comp (g₁ g₂ : β →*o γ) (f : α →*o β) :
+  (g₁ * g₂).comp f = g₁.comp f * g₂.comp f := rfl
+@[to_additive] lemma comp_mul (g : β →*o γ) (f₁ f₂ : α →*o β) :
+  g.comp (f₁ * f₂) = g.comp f₁ * g.comp f₂ := by { ext, exact map_mul g _ _ }
+
+end mul
+end
+
+section ordered_comm_monoid
+variables {hα : ordered_comm_monoid α} {hβ : ordered_comm_monoid β}
+include hα hβ
+
+/-- Helper instance for when there's too many metavariables to apply `to_fun.to_coe_fn` directly. -/
```

Comment:
`to_fun` was an older name for `fun_like`, which apparently stuck around in a couple of docstrings.
```suggestion
/-- Helper instance for when there's too many metavariables to apply `fun_like.to_coe_fn` directly. -/
```

Reply:
Ah great! I was always so confused, but it was in your docstring explaining `fun_like` so I kept it.